[Upload] Documentation for the `upload` response action api (#3520)
elastic/security-docs: docs/management/api/upload-api.asciidoc

[[upload-api]]
=== Upload file to host

> Upload a file to the host running {elastic-defend}.

Reviewer: ```suggestion
Upload a file to a host running {elastic-defend}.
```

> You must have the `File Operations` Security Kibana privilege as part of your role and at least an Enterprise license to perform this action.

Reviewer: ```suggestion
You must have the `File Operations` {kib} privilege in the Security feature as part of your role and at least an Enterprise license to perform this action.
```

==== Request URL

`POST <kibana host>:<port>/api/endpoint/action/upload`

> The request must include:
> 
> * the `Content-Type: multipart/form-data` HTTP header.

Reviewer: If there's only one item in the list, this could be a single sentence:
```suggestion
The request must include the `Content-Type: multipart/form-data` HTTP header.
```

==== Request body

A `multipart/form-data` with the following:

[width="100%",options="header"]
|==============================================
|Name |Type |Description |Required

|`endpoint_ids` |Array (String) |The IDs of endpoints where you want to issue this action. |Yes
|`alert_ids` |Array (String) |If this action is associated with any alerts, they can be specified here. The action will be logged in any cases associated with the specified alerts. |No
|`case_ids` |Array (String) |The IDs of cases where the action taken will be logged. |No

> |`comment` |String |Attach a comment to this action's log. The comment text will appear in associated cases. |No

Reviewer: ```suggestion
|`comment` |String |Attach a comment to this action's history log. The comment text will appear in associated cases. |No
```

> |`parameters.overwrite` |Boolean |Overwrite file on host if it already exists. |No

Reviewer: ```suggestion
|`parameters.overwrite` |Boolean |Overwrite the file on the host if it already exists. |No
```

|`file` |Stream |The file content to be uploaded. |Yes
|==============================================

===== Example requests

> Upload a file named `fix-malware.sh` to a host with an `endpoint_id` value of `ed518850-681a-4d60-bb98-e22640cae2a8`. It assumes that the file is located in the same directory from where the command is being entered:

Reviewer: ```suggestion
Upload a file named `fix-malware.sh` to a host with an `endpoint_id` value of `ed518850-681a-4d60-bb98-e22640cae2a8`. It assumes that the file is located in the same directory where the command is being entered:
```

[source,shell]
--------------------------------------------------
curl -X POST "api/endpoint/action/upload" \
-H 'kbn-xsrf: true' \
-H 'Content-Type: multipart/form-data' \
-F 'endpoint_ids: ["ed518850-681a-4d60-bb98-e22640cae2a8"]' \

> -F "file=@fix-malware.sh" <1>

Reviewer 2: I'm not sure why the console output is reporting an error for this line:

````
15:38:54 INFO:build_docs:asciidoctor: WARNING: management/api/upload-api.asciidoc: line 47: no callout found for <1>
````

The HTML output is rendering correctly and I don't see any problems with the syntax. @elastic/security-docs am I missing something here?

Author: It should be all fixed now @nastasha-solomon . I was trying different options for the language of that code

Reviewer 2: Yep! CI checks passed - thanks!

> -F "file=@fix-malware.sh" <1>

Reviewer 3: Q: Can we also add a regular POST along with the CURL request example like all the other APIs do? 🤔

Author: Not really sure how to do a regular post, since it need to use use `multipart/form-data` 🤷

> -F "file=@fix-malware.sh" <1>
> --------------------------------------------------
> <1> The relative link to the file to be uploaded

Reviewer: Is the `@` symbol part of the required syntax? We might want to make that more explicit, because otherwise it seems unexpected to me, since most key-value pairs use a single character as a connector, like `file=filename.sh` or `file:filename.sh`.

```suggestion
-F "file=@fix-malware.sh" <1>
--------------------------------------------------
<1> The relative path to the file to upload. Note the path must be preceded with `@`.
```
